### ISSUE TITLE
Some questions about the final modifications on RLE loss

### ISSUE THREAD

**Issue by Indigo6** [open-mmlab/mmpose#1363]
@ly015 Hello, thanks for your review and modifications. I have some questions about the final modifications on RLE loss.

1. I wonder whether this method would create a distribution object every time calling backward? I overwrote the _apply() method because of two considerations. First, I think it's not necessary to save constants in state_dict. Second, I thought it would create a distribution object every time backwarding if registering loc and cov to buffer.
https://github.com/open-mmlab/mmpose/blob/e7c7cadbbaf70f8080c093dc2910cd947a2ef6d6/mmpose/models/utils/realnvp.py#L34-L37

2. Why $log\hat\sigma$ is added again？ (⊙ˍ⊙)
https://github.com/open-mmlab/mmpose/blob/e7c7cadbbaf70f8080c093dc2910cd947a2ef6d6/mmpose/models/losses/regression_loss.py#L73-L77

**Comment by ly015** [comment]
1. This modification is for better readability of the code. We prefer not to override the private method of the base class. Your concern about efficiency in distributed environments makes sense. I think we can use properties to return tensors on the proper device instead of buffers:

```python

@property
def prior(self):
    device = next(self.parameters()).device
    loc = torch.zeros(2, device=device)
    cov = torch.eye(2, device=device)
    return distributions.MultivariateNormal(loc, cov)
```

Do you have any suggestions?

2. Although theoretically $log\hat\sigma$ can be removed without affecting the performance, we didn't have enough time to extensively verify it in experiments. So we keep it, for now, to align with the formulation in the paper. We may remove it in the future after verification.

**Comment by Indigo6** [comment]
1. I just tried printing `id(self.prior)` ten times and the results were all the same. Maybe leaving the memory management to python is good enough. 
2. Ok, looking forward to your experiments. (I think, to align with the formulation in the paper, $log\hat\sigma$ should be removed$.)
I have no questions now. Thanks for your reply!
